synapse/lib/cache.py

```diff
@@ -388,7 +388,9 @@ def __init__(self, lookmeth):
 
     def __missing__(self, key):
         valu = self.lookmeth(key)
-        self[key] = valu
+        if valu is not None:
```

Comment:
Cortex uses a `KeyCache` to keep track of sequences which are used by `nexttag`. Unfortunately, if a user calls nexttag on a sequence that does not exist, it will cache None forever and it will not be possible to get the next tag, even if the sequence is created. I'm open to solving the problem differently if we don't want to modify this behavior.

Reply:
This is a bit of a conundrum in general for KeyCache.  Realistically, we almost never want to cache "there was no result".  Perhaps we should modify ( and *def* doc ) the KeyCache behavior to never cache None as an answer?